pyvista/utilities/misc.py

```diff
@@ -17,3 +20,124 @@ class PyvistaDeprecationWarning(Warning):
     """Non-supressed Depreciation Warning."""
 
     pass
+
+
+class VTKVersionInfo():
```

Comment:
Do we really need a full-fledged class for this? Almost all of this seems like a namedtuple, with a custom repr and the API check at the start. A factory that returns a namedtuple seems more concise, even if we have to monkeypatch its `__repr__` (but namedtuples are a hack anyway). Something like
```py
from collections import namedtuple

def VTKVersionInfo():
    version_info = namedtuple('VTKVersionInfo', ['major', 'minor', 'micro'])
    version_info.__repr__ = lambda vi: f"VTK Version {vi.major}.{vi.minor}.{vi.micro}"
    try:
        ver = _vtk.vtkVersion()
        major = ver.GetVTKMajorVersion()
        minor = ver.GetVTKMinorVersion()
        micro = ver.GetVTKBuildVersion()
    except:  # pragma: no cover
        warnings.warn("Unable to detect VTK version. Defaulting to v4.0.0")
        major, minor, micro = (4, 0, 0)
    return version_info(major, minor, micro)
```

This seems to work:
```py
>>> vi = VTKVersionInfo()
>>> vi
VTK Version 9.1.0
>>> vi.major
9
>>> vi.micro
0
>>> 9 in vi  # does this even make any sense?
True
```

Reply:
Oh, and I don't like that bare `except` at all. We should try being more specific; I suspect an `AttributeError` would be appropriate.

Reply:
As I kept adding methods to make it tuple-like I was getting the feeling one of you would be critical about the implementation. I agree that this is quite a bit cleaner. I've removed the `__repr__` monkeypatch because it makes it look more like Python's `sys.version_info`.